### PR TITLE
Fix/button link

### LINKED ISSUE #338: ボタンの当たり判定がバグってる
赤いところ全部クリックできる
buttonをrouter-linkで囲ってるのがそもそもよくないので、BaseButtonの見た目をしたrouter-linkのLinkButtonコンポーネントを別で作るか、BaseButtonにhrefを渡したときはbuttonタグではなくてrouter-linkとして動くようにする
image

### PR DESCRIPTION
**Pull request by ogu-kazemiya**
### **User description**
BaseButtonにto属性をつけたときに、元のBaseButtonと同じ見た目のrouter-linkになるようにした
router-linkとBaseButtonが入れ子になっているところをそれに合わせてすべて修正

close #338

___

### **PR Type**
enhancement

___

### **Description**
- `BaseButton`コンポーネントに`to`属性を追加し、`router-link`を統合しました。
- 各ページで`router-link`を削除し、`BaseButton`を使用してナビゲーションを簡素化しました。
- 不要なCSSクラス`.link`を削除しました。

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>17 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>ContestTeams.vue</strong><dd><code>Replace `router-link` with `BaseButton` for navigation</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/Contest/ContestTeams.vue

<li>Removed <code>router-link</code> and used <code>BaseButton</code> with <code>to</code> attribute.<br> <li> Simplified button structure for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-715a5b5197b6a9ff42072913e621033f3134af98c2a20c6ab3a9252699e60956">+6/-13</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>BaseButton.vue</strong><dd><code>Integrate routing capability into `BaseButton`</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/UI/BaseButton.vue

<li>Added <code>to</code> prop to <code>BaseButton</code> for navigation.<br> <li> Used <code>router-link</code> conditionally based on <code>to</code> prop.<br> <li> Moved <code>.link</code> CSS styling to <code>BaseButton</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-01411dfb5a034777f84eefec0c651e8007e92f81926366c72c9cbd3d034f0aea">+20/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UserProfileDesktop.vue</strong><dd><code>Use `BaseButton` for navigation in UserProfileDesktop</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/User/UserProfileDesktop.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-00724c933f562b5a69e6898067602d1a5a58938f76eafea4463c20da4440b09a">+3/-8</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UserProfileMobile.vue</strong><dd><code>Use `BaseButton` for navigation in UserProfileMobile</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/User/UserProfileMobile.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-73f870cbca3afd6225dd183c9e6139f0255449651aee312b3e6a2d3375a5ab8b">+3/-8</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Contest.vue</strong><dd><code>Simplify navigation using `BaseButton` in Contest page</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Contest.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-d558762d359ad5f4ae0424cef5d1ff91c43f5e01633390e791dc37346b1244a6">+12/-18</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestEdit.vue</strong><dd><code>Use `BaseButton` for navigation in ContestEdit page</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ContestEdit.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-6b345b9b3a019b15846536d31ccad1be7489f3dcc49e2f18453e8701556194be">+6/-15</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestNew.vue</strong><dd><code>Use `BaseButton` for navigation in ContestNew page</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ContestNew.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-2480a3520085ba035bf91cec23e66691c4bb8398c16ea8785b978e50a72ba04e">+8/-14</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestTeamEdit.vue</strong><dd><code>Use `BaseButton` for navigation in ContestTeamEdit page</code>&nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ContestTeamEdit.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-fba1e6a0cebaa7278ce05dc4325ed0db0cdafa561086a629740d23d3ae69d2b3">+5/-9</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestTeamNew.vue</strong><dd><code>Use `BaseButton` for navigation in ContestTeamNew page</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ContestTeamNew.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-0cf527c2206797a81333cbb51d2070441144d1dea8944380cb84cb4516818a10">+5/-9</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Contests.vue</strong><dd><code>Use `BaseButton` for navigation in Contests page</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Contests.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-00aadddd7146b1fe74e665c6f9e4354eb37da680aa61978bd6d579d54dd70605">+6/-9</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Event.vue</strong><dd><code>Use `BaseButton` for navigation in Event page</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Event.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-a575e96bf56c7266c57cdd22bc6606d206114e8988880254045b548acdaa9de0">+8/-14</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Project.vue</strong><dd><code>Use `BaseButton` for navigation in Project page</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Project.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-4e8aa41e439d638b3e4fb2ed6b5fc5fd543a6e78b8e5431f82746141b203122a">+8/-14</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ProjectNew.vue</strong><dd><code>Use `BaseButton` for navigation in ProjectNew page</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ProjectNew.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-02e0c08d10c3b4738460539a64f12e91bdfaaec3367a440492d3e31a4946e5da">+8/-14</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Projects.vue</strong><dd><code>Use `BaseButton` for navigation in Projects page</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Projects.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-363eca605e87df7acfca10ba0acb91ccc250494d61cfbb3c58fdafbc6ca9a82f">+6/-13</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UserAccountEdit.vue</strong><dd><code>Use `BaseButton` for navigation in UserAccountEdit page</code>&nbsp; &nbsp; </dd></summary>
<hr>

src/pages/UserAccountEdit.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-6dd39e0ccbeb6480292f207d72db2c009a37026942f2c4a8082d94037f5e2494">+8/-14</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UserAccountNew.vue</strong><dd><code>Use `BaseButton` for navigation in UserAccountNew page</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/UserAccountNew.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-0e1077c6ce1d02539bd752d0b681d55167c2d645ce8deaf4ceae65eb1ba65de9">+8/-14</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UserAccounts.vue</strong><dd><code>Use `BaseButton` for navigation in UserAccounts page</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/UserAccounts.vue

<li>Replaced <code>router-link</code> with <code>BaseButton</code> for navigation.<br> <li> Removed unnecessary CSS class <code>.link</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/353/files#diff-8fa659e3153dfa9e05cfc9887d7bce3d6fb7562066f065335a501f193c5e4c69">+13/-17</a>&nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by Pugma** [comment]
これは僕個人の意見なんだけど、一つのファイルにまとめてしまうよりも、`BaseLink.vue` というファイルを新しく作って完全に機能を分離したほうが分かりやすいかもな〜って思いました

今だとリンクとして置きたくても、例えば `:to` 属性を設定し忘れば自動でボタンになっちゃいます
こういうときに、リンク用でファイルを分けて、リンクの方には `:to` 属性を必須にしておけばミスも減るんじゃないかな〜って思いました

**Comment by ogu-kazemiya** [comment]
なるほど
どっちにするか迷ってたんですが、分けたほうが良さそうですね

通常のリンクと区別したいので、コンポーネント名は`LinkButton`にします

File: src/components/UI/BaseButton.vue [hunk 1]
```diff
@@ -29,7 +29,7 @@ const props = withDefaults(defineProps<Props>(), {
 
 <style lang="scss" module>
 .button {
-  display: flex;
+  display: inline-flex;
```
Comment (mehm8128) [inline src/components/UI/BaseButton.vue:32]: この差分不要になってそうですかね？不要だったら戻しておいてほしいです 🙏

File: src/pages/Contest.vue [hunk 2]
```diff
@@ -73,15 +72,14 @@ const contestTeams: ContestTeam[] = (
...
-      </base-button>
-    </router-link>
+    <link-button
+      :to="{ name: 'Contests' }"
+      :class="$style.backButton"
+      type="secondary"
+      icon="mdi:arrow-left"
+    >
+      Back
+    </link-button>
```
Comment (mehm8128) [inline src/pages/Contest.vue:82]: `LinkButton.vue`で`inline-flex`を`flex`にした影響で、divで囲っていないところが伸びちゃってそうです(もしかしたら他のページもかも)
image

Comment (ogu-kazemiya) [reply src/pages/Contest.vue:82]: 揃えるためにと思ったのですが、そもそもこのissueで問題になっていたのはそこでした…
LinkButtonのほうのdisplayは戻しておきます

**Review by mehm8128 — APPROVED** [review]
よさそうです！:kansya:

**Pull request merged**